src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -70,9 +70,18 @@ public sealed partial class QuicStream
     {
         CancellationAction = target =>
         {
-            if (target is QuicStream stream)
+            try
```

Comment:
I'm wondering if we can have private version of `stream.Abort` that does not throw and either is silent or returns `bool` for tracing if we ant to know if `Abort` succeeded or not.

Reply:
Abort already tries to be no-op in case stream is disposed. But it is not enough. The only way to fix that I can think of, is to do dangerousAddRef in the beginning of Abort, and only after that check _disposed. Or wrap both msquic abort call and dispose of the handle in the lock and check _disposed inside the lock before abort. Otherwise, by the time we access handle on msquic abort call, it might get disposed already, despite the check in the beginning of Abort.

Reply:
So the ODE would come from the SafeHandle? I'm fine with this change for 7.0. It seems to cause lot of CI troubles. We can figure later if there is cheaper way to do it.

Reply:
The `ObjectDisposedException` is coming from `DangerousAddRef`. At the moment I see no way of avoiding the exception without a lock (we don't have `TryAddRef` or similar). I expect the exception to be relatively rare so I think it is fine (at least for now)